README.md

```diff
@@ -46,3 +46,72 @@ this project took
 * advanced option handling
 * `%autosetup`/`%autopatch`
 * many more nuts and bolts under the hood
+
+## how to install
+
+debbuild's packages are (naturally) built by debbuild itself. You can generate an installable
+`.deb` file for your distribution by running the `./bootstrap.sh` script.
+
+There will typically be prebuilt packages for the latest Ubuntu LTS release attached to GitHub
+releases, but you will find packages for other Debian and Ubuntu releases on the openSUSE Build
+Service:
+
+* Debian: https://download.opensuse.org/repositories/Debian:/debbuild/
+* Ubuntu: https://download.opensuse.org/repositories/Ubuntu:/debbuild/
+
+## how to use
+
+Much like `rpmbuild`, debbuild requires a certain directory structure under your home
+directory to build debs. You can easily do this with the following one-line command:
+
+```
+mkdir -p $HOME/debbuild/{BUILD,DEBS,SOURCES,SPECS,SDEBS}
+```
+
+Place the `.spec` file in `$HOME/debbuild/SPECS`, and the source tarball, named in the format
+of `<Name>-<Version>.tar.gz`, in `$HOME/debbuild/SOURCES`. Execute debbuild with the command
+`debbuild -bb $HOME/debbuild/SPECS/package.spec` and if all goes well it will build a `.deb`
+file and output it to `$HOME/debbuild/DEBS/$arch` (where `$arch` is likely `all` or the name of
+your machine's architecture, like `amd64` or `arm64`).
+
+### helpful pastes
+
+If you just need to do a quick-and-dirty build of a project you cloned from git, the following
+pastes might help. Please adapt them to your needs.
+
+This assumes you have checked the project out to a directory that matches the package name.
+
+The following will create a source archive from the most recent Git tag, update the `Version`
+field in the specfile to match, and build a deb:
+
+```
+PROJECT=$(basename `pwd`); \
+TAG=$(git tag --sort=version:refname | tail -1); \
+SPEC=$(find . -type f -name \*.spec | head -1); \
+git archive --format=tgz --prefix=${PROJECT}-${TAG}/ -o $HOME/debbuild/SOURCES/${PROJECT}-${TAG}.tar.gz $TAG && \
+cp -v "$SPEC" "$HOME/debbuild/SPECS/$(basename $SPEC)" && \
+sed -re "s/^Version: (.*)$/Version: $TAG/" -i "$HOME/debbuild/SPECS/$(basename $SPEC)" && \
+debbuild -bb "$HOME/debbuild/SPECS/$(basename $SPEC)"
+```
+
+The following does the same, but it uses the number of commits in the current branch as a
+version number instead:
+
+
+```
+PROJECT=$(basename `pwd`); \
+REV=$(git rev-list --count HEAD); \
+SPEC=$(find . -type f -name \*.spec | head -1); \
+git archive --format=tgz --prefix=${PROJECT}-${REV}/ -o $HOME/debbuild/SOURCES/${PROJECT}-${REV}.tar.gz HEAD && \
+cp -v "$SPEC" "$HOME/debbuild/SPECS/$(basename $SPEC)" && \
+sed -re "s/^Version: (.*)$/Version: $REV/" -i "$HOME/debbuild/SPECS/$(basename $SPEC)" && \
```

Comment:
Wouldn't it make sense to make this something like: `$TAG+git$NUM` where `$TAG` is latest tagged version and `$NUM` is number of commits since that tag?